src/mono/mono/metadata/class-setup-vtable.c

```diff
@@ -773,6 +773,13 @@ mono_method_get_method_definition (MonoMethod *method)
 static gboolean
 verify_class_overrides (MonoClass *klass, MonoMethod **overrides, int onum)
 {
+	// on windows and arm, we define NDEBUG for release builds
+	// on browser and wasi, we define DEBUG for debug builds
+#if defined(NDEBUG) || !defined(DEBUG)
+	if (klass->image == mono_defaults.corlib)
+		return TRUE;
+#endif
```

Comment:
I would prefer identical behavior in checked and debug builds. Otherwise it's easy to forget about this and end up investigating an issue on a runtime that can't reproduce it.

Maybe a debug build can do the slow check and emit a warning if the expected fast answer doesn't match the slow check.

Reply:
Once you get a chance to go over it more thoroughly, I'll change it to be based on ENABLE_CHECKED_BUILD. I'll also add validation in checked builds to the caching path.